### ISSUE TITLE
AllocCSEntity does not check whenever entity is valid

### ISSUE THREAD

**Issue by DBotThePony** [Foohy/jazztronauts#91]
It only check if entity *ever* existed in pool 
https://github.com/Foohy/jazztronauts/blob/c56b7a594283c13391d4f8b4238fc0c290a62955/gamemodes/jazztronauts/gamemode/lib/sh_csent.lua#L62
```
gamemodes/jazztronauts/gamemode/ui/dialog/sh_dialogcmds.lua:114: Tried to use a NULL entity!
	 1: (message = "gamemodes/jazztronauts/gamemode/ui/dialog/sh_dialogcmds.lua:114: Tried to use a NULL entity!") [gcompute/execution/local/gluaexecutioninstance.lua: 178]
	 2: SetPos ([NULL CSEntity], -6208.070313 -1221.079224 1165.608154, "Tried to use a NULL entity!") [[C]: -1]
	 3: _PPM2_CreatePlayerProxy () [gamemodes/jazztronauts/gamemode/ui/dialog/sh_dialogcmds.lua: 114]
	 4: CreatePlayerProxy () [[VLL2:VM:ppm2:ppm2/client/hooks.lua]: 233]
	 5: () [@repl_14: 1]
	 6: xpcall (function (message)) [[C]: -1]
	 7: Start (self = { GCompute.Execution.GLuaExecutionInstance: 0x01ece6b1b8a0 }) [gcompute/execution/local/gluaexecutioninstance.lua: 173]
	 8: CreateExecutionInstance (self = { GCompute.Execution.GLuaExecutionContext: 0x01ecf27c7560 }, code = "TEST_MODEL = dialog.CreatePlayerProxy()", sourceId = "@repl_14", instanceOptions = 6, callback = nil) [gcompute/execution/executioncontext.lua: 56]
	 9: (self = { GCompute.IDE.ViewTypes.Types.Console.Constructor: 0x01ecea75fb78 }, code = "TEST_MODEL = dialog.CreatePlayerProxy()") [gcompute/ui/ide/views/console/console.lua: 372]
	10: xpcall (GLib.Error) [[C]: -1]
	11: () [glib/threading/thread.lua: 262]
```

![gmod_12-48-53-9505825](https://user-images.githubusercontent.com/6339511/71303792-40880280-23f0-11ea-9b24-a046ea2891e0.png)

**Comment by Commenter25** [comment]
This has been resolved, and should be closed. (sorry for spamming you in particular dbot, you just left a lot of issues that didn't get closed xd)

**Comment by Foohy** [comment]
Fixed in #120! Closing out (ditto to you Commenter thank you tons for going through these issues)
